### PR TITLE
better compilation across the board

### PR DESCRIPTION
**Pull request by JerrySievert**
more travis fixes, as well as OS X build fixes

### CODE REVIEW

**Comment by JerrySievert** [comment]
@dacjames, @car3ca can i convince you both to take a look at this, and make sure it works for you? i want to use this as the new base for getting fixes in.

what i still haven't managed to fix is 9.2, 9.3 tests - for some reason, the makefile checks aren't being honored on travis (they are honored on the same platform locally), and jsonb tests are still being run.

if either of you have ideas, i'd love to hear them.

thanks!

**Comment by car3ca** [comment]
I'll look at it during this week

**Comment by dacjames** [comment]
Yes, I can probably get to it tonight, definitely this weekend.

**Comment by JerrySievert** [comment]
great, thanks both of you!

now, if only i could track down why the jsonb tests are running in 9.2/9.3 on travis, but not locally on the same platform.

**Comment by ruimarinho** [comment]
Just tested make static on 9.5.1 / OSX 11.11.3:

```
❯ make installcheck
cat: stdout: Broken pipe
/usr/local/lib/postgresql/pgxs/src/makefiles/../../src/test/regress/pg_regress --inputdir=./ --bindir='/usr/local/Cellar/postgresql/9.5.1/bin'    --dbname=contrib_regression init-extension plv8 plv8-errors inline json startup_pre startup varparam json_conv jsonb_conv window guc es6 arraybuffer dialect
(using postmaster on Unix socket, default port)
============== dropping database "contrib_regression" ==============
NOTICE:  database "contrib_regression" does not exist, skipping
DROP DATABASE
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== running regression test queries        ==============
test init-extension           ... ok
test plv8                     ... ok
test plv8-errors              ... ok
test inline                   ... ok
test json                     ... ok
test startup_pre              ... ok
test startup                  ... ok
test varparam                 ... ok
test json_conv                ... ok
test jsonb_conv               ... ok
test window                   ... ok
test guc                      ... ok
test es6                      ... ok
test arraybuffer              ... ok
test dialect                  ... ok

======================
 All 15 tests passed.
======================
```

**Comment by car3ca** [comment]
The following pull request https://github.com/plv8/plv8/pull/147 should fix the problems for versions under 9.4.

File: Makefile [hunk 1]
```diff
@@ -45,32 +45,37 @@ endif
...
-	ifeq ($(GCCVERSION),4.6)
-		OPTFLAGS = -O2 -std=gnu++0x -fno-rtti
-	else
-		OPTFLAGS = -O2 -std=c++11 -fno-rtti
-	endif
-endif
+# for older g++ (e.g. 4.6.x), which do not support c++11
+#OPTFLAGS = -O2 -std=gnu++0x -fno-rtti
+
+OPTFLAGS = -O2 -std=c++11 -fno-rtti
```
Comment (dacjames) [inline Makefile]: What is the reason for the -fno-rtti?

**Comment by dacjames** [comment]
Works here.

```sh
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== running regression test queries        ==============
test init-extension           ... ok
test plv8                     ... ok
test plv8-errors              ... ok
test inline                   ... ok
test json                     ... ok
test startup_pre              ... ok
test startup                  ... ok
test varparam                 ... ok
test json_conv                ... ok
test jsonb_conv               ... ok
test window                   ... ok
test guc                      ... ok
test es6                      ... ok
test arraybuffer              ... ok
test dialect                  ... ok

======================
 All 15 tests passed.
======================
```

File: Makefile (hunk 1 above)
Comment (JerrySievert) [inline Makefile]: it was part of the original makefile for older versions of gcc - assuming it was meant to try to lower the library size.

**Pull request merged**